commit/outcome_test.go

```diff
@@ -42,3 +44,81 @@ func Test_buildReport(t *testing.T) {
 		}
 	})
 }
+
+func Test_fChainConsensus(t *testing.T) {
+	lggr := logger.Test(t)
+	f := 3
+	fChainValues := map[cciptypes.ChainSelector][]int{
+		cciptypes.ChainSelector(1): {5, 5, 5, 5, 5},
+		cciptypes.ChainSelector(2): {5, 5, 5, 3, 5},
+		cciptypes.ChainSelector(3): {5, 5},             // not enough observations, must be observed at least f times.
+		cciptypes.ChainSelector(4): {5, 3, 5, 3, 5, 3}, // both values appear at least f times
+	}
+	fChainFinal := fChainConsensus(lggr, f, fChainValues)
+
+	assert.Equal(t, map[cciptypes.ChainSelector]int{
+		cciptypes.ChainSelector(1): 5,
+		cciptypes.ChainSelector(2): 5,
+		cciptypes.ChainSelector(4): 3,
```

Comment:
Why pick 3 instead of 5?

Reply:
Isn't this case impossible since the threshold is more than half of the committe?

Reply:
IMO this should be an error. There is no consensus about what the true value is.

Reply:
We have 2 implementations for how to merge observations together. If you were to use [MinObservationFilter](https://github.com/smartcontractkit/chainlink-ccip/blob/ccip-develop/execute/internal/validation/reports.go) I think it would return that there are 2 valid values. Since fChain is a single number that would be an error.

Reply:
So seems better to treat this case as "not reaching consensus". I pushed a fix